dags/common/enricher.py

```diff
@@ -0,0 +1,109 @@
+import os
+import re
+import xml.etree.ElementTree as ET
+from typing import Dict
+
+import backoff
+import requests
+from structlog import get_logger
+
+
+class Enricher(object):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_schema(self):
+        return os.getenv("REPO_URL", "http://repo.qa.scoap3.org/schemas/hep.json")
+
+    def _clean_arxiv(self, arxiv):
+        if arxiv is None:
+            return None
+        return re.findall(r"(\d{4}\.\d{4,5})", arxiv)[0]
```

Comment:
🙃 this will fail if doesn't find a match, for example:

```

In [0]: re.findall(r"(\d{4}\.\d{4,5})", '2222.000')[0]                                                                                                                                                                                                                                                                                                            
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
```

we could do something like:

```python
try:
    return re.match(r"(\d{4}\.\d{4,5})", '2222.000').group() 
except AttributeError:
    return None
```

Reply:
Yes good point, updated